File: typecast-java/src/main/java/io/typecast/exceptions/RateLimitException.java
package io.typecast.exceptions;

/**
 * Exception thrown when the API returns a 429 Too Many Requests error.
 * 
 * <p>This indicates the rate limit has been exceeded.</p>
 */
public class RateLimitException extends TypecastException {
    
    /**
     * Creates a new RateLimitException.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     */
    public RateLimitException(String message, String responseBody) {
        super(message, 429, responseBody);
    }

    /**
     * Creates a new RateLimitException with a cause.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     * @param cause        the underlying cause
     */
    public RateLimitException(String message, String responseBody, Throwable cause) {
        super(message, 429, responseBody, cause);
    }
}

File: typecast-php/src/Exceptions/RateLimitException.php
<?php

declare(strict_types=1);

namespace Neosapience\Typecast\Exceptions;

class RateLimitException extends TypecastException
{
    public function __construct(string $message = 'Rate limit exceeded', ?\Throwable $previous = null)
    {
        parent::__construct($message, 429, $previous);
    }
}
